### PR TITLE
Create Login Page

### PR DESCRIPTION
**Pull request by faysal-alam-sazim**
### Description of Change
- Developed a login page
- Used validation for login inputs
- Password validation is added: Must include one uppercase, one lowercase and one special character
- Password must be minimum 6 character long
- After successful login, redirects to todo app page 
### Associated Ticket Link:
- https://trello.com/c/pxUG43fG
### Screenshots / Screen Recordings: 

https://github.com/user-attachments/assets/a9e261a5-37fa-4339-b4c9-cc956224a915

### CODE REVIEW

File: pages/login/index.page.tsx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import LoginPageContainer from "@/modules/LoginPage/containers/LoginPageContainer";
```
Comment (ragib-hossain-sazim) [inline pages/login/index.page.tsx:1]: `LoginPageContainer` => `LoginContainer`. Please refrain from using the `Page` word from your page, container and component names

**Comment by ragib-hossain-sazim** [comment]
[JC] Why do we have 2 commits with same name? Are both these commits doing the same thing?

**Comment by faysal-alam-sazim** [comment]
> [JC] Why do we have 2 commits with same name? Are both these commits doing the same thing?

I forgot that I've already commited before. So I didn't use --amend on second commit.

File: modules/LoginPage/components/LoginForm/LoginForm.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  Box,
+  Button,
+  PasswordInput,
+  Text,
+  TextInput,
+  Title,
+} from "@mantine/core";
+
+import classes from "./LoginForm.module.css";
+import { TLoginCredentials } from "@/shared/typedefs/types";
```
Comment (ishmam-iqbal-sazim) [inline modules/LoginPage/components/LoginForm/LoginForm.tsx:14]: @ and relative imports should be separate.

File: modules/LoginPage/containers/LoginContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import { Box, Flex, Image } from "@mantine/core";
+
+import LoginForm from "../components/LoginForm/LoginForm";
+
+function LoginContainer() {
```
Comment (ishmam-iqbal-sazim) [inline modules/LoginPage/containers/LoginContainer.tsx:6]:
```suggestion
const LoginContainer = () => {
```
Please create a ticket to convert all component definitions into arrow functions.

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comments.

**Review by ragib-hossain-sazim — APPROVED** [review]

**Pull request merged**